test/gold.sh

```diff
@@ -22,10 +22,10 @@ source test/common.sh  # for $OSH
 _compare() {
   set +o errexit
 
-  "$@" >_tmp/shebang.txt
+  PATH="$PWD/_tmp/gold-bin:$PATH" "$@" >_tmp/shebang.txt
```

Comment:
revert all changes to test/gold.sh please :)  The new test is in gold/readlink.sh